Zend/zend_builtin_functions.stub.php

```diff
@@ -6,12 +6,17 @@ class stdClass
 {
 }
 
+/** @refcount 1 */
 function zend_version(): string {}
 
 function func_num_args(): int {}
 
 function func_get_arg(int $position): mixed {}
 
+/**
+ * @return array<int, mixed>
```

Comment:
I wonder whether it would make sense to distinguish `mixed[]` and `array<int, mixed>`, where the former requires a packed array, while the latter could be a hash with integer keys. We can actually distinguish this in opcache (MAY_BE_ARRAY_PACKED vs MAY_BE_ARRAY_NUMERIC_HASH) though we currently don't for func info...

Reply:
Though upon further consideration, there may be issues here with interned empty arrays, which are not packed.

Reply:
in psalm/phpstan terms, there is also a `list<mixed>` which might be a precise type for this case

see e.g. https://psalm.dev/docs/annotating_code/type_syntax/array_types/#lists

Reply:
> in psalm/phpstan terms, there is also a list<mixed> which might be a precise type for this case

Yeah, `list` sounds like a great name, but I don't like the fact that it clashes with a hypothetical `list` class name.

Alternatively, I could also imagine using the comment section of the `@return` annotation for providing additional meta information. Something like: `@return array<int, mixed> packed, non-reference`.